tests/FakeItEasy.IntegrationTests/ExternalAssemblyGenerator.cs

```diff
@@ -104,7 +104,7 @@ public class Foo { }
             var emitResult = compilation.Emit(this.AssemblyDependencyPath);
             if (!emitResult.Success)
             {
-                throw new Exception("Failed to create assembly - " + emitResult.Diagnostics[0]);
+                throw new InvalidOperationException("Failed to create assembly - " + emitResult.Diagnostics[0]);

```

Comment:
Many instances of `Exception` being too generic. I struggled, but picked some specific ones.

Reply:
Me, I would have just disabled the warning for the tests...

Reply:
Oh, I can change to disable globally within tests if you prefer.

Reply:
Sounds good to me.